### PR TITLE
Javascript-II Solutions

### PR DESCRIPTION
**Pull request by lex-node**
includes solutions to all problems and stretch problems, except for Challenge-3 in closures.js (still working on that one but I suspect I need to let closures marinate in my brain soup for a few more days before I can crack it)

### CODE REVIEW

File: assignments/callbacks.js [hunk 1]
```diff
@@ -26,30 +27,68 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
...
+    return cb(list.includes(item));
 }
 
+contains('Pencil', items, (boolValue) => {
+    console.log(boolValue);
+})
+
+contains('Pickle', items, (boolValue) => {
+    console.log(boolValue);
+})
```
Comment (jacobAngulo) [inline assignments/callbacks.js:77]: one note just starting off, for the case of keeping it DRY, with the way you used your callbacks you could have made one callback `const logger = value => console.log(value)` and applied that everywhere

File: assignments/callbacks.js [hunk 2]
```diff
@@ -26,30 +27,68 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
...
+// Pass the duplicate free array to the callback function.
+// Do not mutate the original array.
+
+const itemsDupes = ['Pencil', 'Notebook', 'Pencil', 'yo-yo', 'Gum'];
+
 function removeDuplicates(array, cb) {
-  // removeDuplicates removes all duplicate values from the given array.
-  // Pass the duplicate free array to the callback function.
-  // Do not mutate the original array.
+    return cb(array.filter((item, index) => array.indexOf(item) === index));
```
Comment (jacobAngulo) [inline assignments/callbacks.js:89]: would this work with three duplicates?

**Review by jacobAngulo — APPROVED** [review]
good work all in all! looks like you're pretty comfortable with everything, lemme know if you need any help

File: assignments/callbacks.js (hunk 1 above)
Comment (lex-node) [reply assignments/callbacks.js:77]: Makes sense. More elegant!

**Pull request merged**